### ISSUE TITLE
Issue scanning 'newer' Driver Licences

### ISSUE THREAD

**Issue by pjdupreez** [marcus-bornman/rsa_scan#2]
Awesome package. Couldnt get others to work, but this one works really fast. I did however have to specify the barcode type to scan (dont even knownif it is worth a pull request, very small change).

What I have noticed though, is some Drivers Licences scan immediately, whilest others don't.

I found that the 'newer' barcodes are printed poorly. The Start and Stop Pattern are broken and this seems to cause issues.

I have uploaded a side by side comparison of my old licence and 'new' one (about 3 years old). I also checked some friends' and family members' licences. Same thing. All the new ones have broken lines and wont scan.

Any advise?

![20210412_161811.jpg](https://user-images.githubusercontent.com/1033864/114409415-bba79600-9baa-11eb-8e99-ccf82208bee8.jpg)

**Comment by pjdupreez** [comment]
I managed to fix this by keeping the flash on while scanning (_cameraController.setFlashMode(FlashMode.torch). 

It seems that lighting has a major effect on the latency of scans.

**Comment by the-mars-rover** [comment]
Hi @pjdupreez :) Sorry for the delayed response - thought I'd provide some insight even though the issue has been closed.

So, we're quite reliant on the flutter `camera` package, which is still in a tech-preview state. We're currently depending on an older version of the plugin. I have updated to a newer version of the plugin but am yet to check-in or release the change because, unfortunately, I want to wait for the following PR to go in on the firebase_ml_vision plugin as well: https://github.com/FirebaseExtended/flutterfire/pull/2368

Hopefully we can move over to the newer versions of these plugins soon, which should improve the quality of image streams (for which some notable changes have gone in recently) and, consequently, the performance of scanning for lower quality barcodes.

So, in short, keep an eye on the mentioned PR and maybe give it a thumbs up :)

**Comment by pjdupreez** [comment]
Will do, thanks

**Comment by the-mars-rover** [comment]
@pjdupreez I've also logged a new issue for the dependency upgrades: https://github.com/born-ideas/rsa_scan/issues/3
